packages/basti-cdk/src/basti-access-security-group.ts

```diff
@@ -0,0 +1,65 @@
+import { aws_ec2 } from 'aws-cdk-lib';
+
+import { generateShortId } from './basti-helper';
+import { TARGET_ACCESS_SECURITY_GROUP_NAME_PREFIX } from './basti-constants';
+
+import type { Construct } from 'constructs';
+import type { BastiInstance } from './basti-instance';
+
+/**
+ * The properties for the bastion access security group.
+ */
+export interface BastiAccessSecurityGroupProps
+  extends aws_ec2.SecurityGroupProps {
+  /**
+   * Basti ID
+   *
+   * @default - A 8-character pseudo-random string
+   */
+  readonly bastiId?: string;
+}
+
+/**
+ * The security group for the bastion instance.
+ */
+export class BastiAccessSecurityGroup extends aws_ec2.SecurityGroup {
+  /**
+   * Basti ID
+   */
+  readonly bastiId: string;
+
+  /**
+   * Constructs a new instance of the BastiAccessSecurityGroup class.
+   * @param scope The scope of the construct.
+   * @param id The ID of the construct.
+   * @param props The properties of the construct.
+   */
+  constructor(
+    scope: Construct,
+    id: string,
+    props: BastiAccessSecurityGroupProps
+  ) {
+    const bastiId = props.bastiId ?? generateShortId(id);
+    super(scope, id, {
+      securityGroupName: `${TARGET_ACCESS_SECURITY_GROUP_NAME_PREFIX}-${bastiId}`,
+      vpc: props.vpc,
+      allowAllOutbound: true,
+    });
+
+    this.bastiId = bastiId;
+  }
+
+  /**
+   * Adds an ingress rule to the security group. That allows the
+   * bastion instance to access the target instance.
+   *
+   * @param bastiInstance The Basti instance
+   * @param port The port to allow access to
+   */
+  public addBastiInstance(
+    bastiInstance: BastiInstance,
```

Comment:
I'm not entirely sure what you mean by this. Could you please explain a bit more what you mean by lookup?

Reply:
@bobveringa, I meant something like the [VPC.fromLookup](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Vpc.html#static-fromwbrlookupscope-id-options) method. Such a method will be useful when the `BastiInstance` is defined in another CDK application and the user only has its `bastiId`

Reply:
Ah yeah, that is a good idea. But it might be a bit difficult to implement. The BastiInstance construct creates 3 resources:
1. ec2.Instance
2. iam.Role
3. ec2.SecurityGroup

The resources `iam.Role` and `ec2.SecurityGroup` both allow the creation of their interface counterparts (`ISecurityGroup` and `IRole`) using the `.fromName` (or similarly named) method. However, it seems like `Instance` does not support any `.form...` methods. There have been requests for this (see https://github.com/aws/aws-cdk/issues/3911) but it seems like AWS is not interested in supporting it.

That leaves us with a few options:
1. We support it partially, we only look up both the security group and role from the ID but do not import the instance. This feels a bit wrong, but it would support this use case. 
2. Fully implement the `instance.fromLookup()` method and get it pushed in AWS. This would likely make more people happy, but there is probably a reason they didn't implement this, which means there must be something complex.
3. Wait for CDK support? This would likely take a while, since that issue I linked is from 2019.
4. Use a workaround.There is a workaround available (it is not great, but it is there):
All that the `bastiInstance` is used for here is the security group it contains. And given that `BastiAccessSecurityGroup` is essentially just a regular `SecurityGroup` but with some fluff. Users could export the name/id of the `basti.securityGroup` and just add an ingress line for that group at the given port. It is not as neat but achieves the same purpose.

None of these options are particularly great, 1. Feels not great because if we implement something like `BastiInstance.fromId()` and it then doesn't provide the basti instance, then that's no good. 2 and 3 are unlikely to happen (unless someone is brave enough to do them). And 4 is a workaround that is not that pretty.

If you feel like that, this functionality is vital for release than we can certainly implement this. But it doesn't add that much functionality, and there is a workaround that is available for people that have set-ups like that. If people really need this functionality we can revisit it.

Reply:
I think the first option is not bad at all 👍 

We could add a limited `IBastiInstance` interface and a corresponding implementation that would be returned from the `BastiInstance.fromBastiId` method. This limited implementation will be usable in the `BastiAccessSecurityGroup. addBastiInstance` method and maybe in other use cases that we haven't thought of as well.

Later, as AWS adds the `from...()` method for EC2 instances, we could expand the `IBastiInstance` interface.

I think it would be great to have it right from the first version if it is not too hard to implement.